src/main/java/gift/service/ProductService.java

```diff
@@ -9,32 +12,47 @@
 
 @Service
 public class ProductService {
+
     private final ProductRepository productRepository;
+    private final CategoryService categoryService;
 
-    public ProductService (ProductRepository productRepository){
+    public ProductService(ProductRepository productRepository, CategoryService categoryService) {
         this.productRepository = productRepository;
+        this.categoryService = categoryService;
     }
+
     @Transactional
-    public Product save (Product product){
-        return productRepository.save(product);
+    public Product save(ProductRequestDto productRequestDto) {
+        Category category = categoryService.findById(productRequestDto.getCategoryId());
+        return productRepository.save(productRequestDto.toEntity(category));
     }
 
-    @Transactional(readOnly=true)
-    public List<Product> findAll(){
+    @Transactional(readOnly = true)
+    public List<Product> findAll() {
         return productRepository.findAll();
     }
 
     @Transactional(readOnly = true)
-    public Product findById(Long id) throws ProductNotFoundException{
+    public Product findById(Long id) throws ProductNotFoundException {
         return productRepository.findById(id).get();
     }
 
     @Transactional
-    public void updateProduct(Long id,Product product) throws ProductNotFoundException{
-        productRepository.save(new Product(id,product.getName(),product.getPrice(),product.getImageUrl()));
+    public void updateProduct(Long id, ProductRequestDto productRequestDto) {
+        Product product = productRepository.findById(id)
+            .orElseThrow(() -> new ProductNotFoundException(
+                Messages.NOT_FOUND_PRODUCT_MESSAGE));
+        Category updatedcategory = categoryService.findById(productRequestDto.getCategoryId());
+        product.setName(productRequestDto.getName());
+        product.setPrice(productRequestDto.getPrice());
+        product.setImageUrl(productRequestDto.getImageUrl());
+        product.setCategory(updatedcategory);
```

Comment:
set 메서드 대신에 아래와 같이 하나의 메서드로 표현해보면 어떨까요?
```suggestion
product.update(productRequestDto.getName(), productRequestDto.getPrice(), productRequestDto.getImageUrl(), updatedCategory);
```
setter 사용도 안할 수 있고 변경 기능을 한곳에서 제공할 수 있을 것 같아요.